src/components/home/Home.js

```diff
@@ -42,40 +46,47 @@ function Home() {
   const { t } = useTranslation();
   const classes = useStyles();
   const [searchResults, setSearchResults] = useState(null);
+  const [searchInput, setSearchInput] = useState(null);
+  const [range, setRange] = useState(RANGES.ALL.value);
+  const [keywords, setKeywords] = useState(null);
   const { hooks } = useContext(QueryClientContext);
-  const {
-    data: collections,
-    isLoading,
-    isPlaceholderData,
-  } = hooks.usePublicItemsWithTag(PUBLISHED_TAG_ID, {
-    placeholderData: PLACEHOLDER_COLLECTIONS,
-  });
-  const { data: members } = hooks.useMembers(
-    null,
-    isPlaceholderData
-      ? null
-      : [...new Set(collections?.map(({ creator }) => creator).toArray())],
+  const { data: collections, isLoading } = hooks.usePublicItemsWithTag(
+    PUBLISHED_TAG_ID,
+    {
+      placeholderData: PLACEHOLDER_COLLECTIONS,
+      withMemberships: true,
+    },
   );
+
   const collectionsGraasper = collections?.filter(
     (collection) => collection.creator === NEXT_PUBLIC_GRAASPER_ID,
   );
 
-  const handleSearch = (event) => {
-    const query = event.target.value.trim().toLowerCase();
-    if (query.length > 0) {
-      setSearchResults(
-        collections.filter(
-          (collection) =>
-            collection.name.toLowerCase().includes(query) ||
-            members
-              ?.find(({ id }) => collection.creator === id)
-              ?.name.toLowerCase()
-              .includes(query),
-        ),
-      );
+  const { data: resultCollections } = hooks.useKeywordSearch(range, keywords);
+
+  useEffect(() => {
+    setSearchResults(resultCollections);
+  }, [resultCollections]);
+
+  const handleKeywordInput = (event) => {
+    setSearchInput(event.target.value.trim().toLowerCase());
+  };
+
+  const handleClick = () => {
+    if (searchInput.split(' ').length > 1) {
+      const keywordArray = searchInput
+        .split(' ')
+        .map((keyword) => keyword.trim());
+      setKeywords(keywordArray.join('&'));
+    } else {
+      setKeywords(searchInput);
```

Comment:
I might be wrong, but probably you don't even need to check for an array, the array process should work on a string as well. The only difference I see is that it won't send a string anymore, but it doesn't matter, does it?

Reply:
Yes I don't need to check that.
I added a check for range, because in case of title, the input will be considered a single string. The main reason is in case of title, the backend uses `LIKE '%keyword%` , which I think works better than full text search in case of title